config/api/assessment_schema.json

```diff
@@ -0,0 +1,382 @@
+{
+  "id": "#{Rails.root}/config/api/assessment_schema.json",
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Legal Aid Check Financial Eligibility payload schema",
+  "description": "This schema defines the payload required to submit data to the Legal Aid Check Financial Eligibility API",
+
+
+  "type": "object",
+  "required": [
+    "assessment_id",
+    "meta_data",
+    "applicant",
+    "applicant_income",
+    "applicant_outgoings",
+    "applicant_capital"
+  ],
+  "additionalProperties": false,
+
+
+  "definitions": {
```

Comment:
I think `$id` can be used here too to clean up some of the uses of these core objects. For example:
```
"uuid": {
      "$id": "#uuid",
      "type": "string",
      "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$",
      "description": "has to be a valid uuid"
    }
```
Then where a uuid is needed instead of:
```
"$ref": "#/definitions/uuid"
```
We can do:
```
"$ref": "#uuid"
```
Again see - http://json-schema.org/understanding-json-schema/structuring.html#the-id-property

Reply:
Again, not supported in draft 4.

Reply:
Fair enough.